packages/admin/cms-admin/src/generator/future/generateForm.ts

```diff
@@ -99,7 +102,9 @@ export function generateForm(
         gqlIntrospection,
         baseOutputFilename,
         fields: config.fields,
+        fragmentName,
         formConfig: config,
+        gqlType: config.gqlType,
```

Comment:
here you can use formConfig.gqlType for sure

Reply:
do you mean inside of generateFormFields and generateFormLayout? I initially tried that but that didn't work because optionalFields works nested. So it's required for optionalFields.fields to check the nested graphql-type to find the correct introspection-data. (it's not in the pull-request changes because I reused the already existing var. It's used [here](https://github.com/vivid-planet/comet/pull/2352/files#diff-0e37c98d83368ee5018182dd50293b61c77ca082054221927955e49675ebf3aaR36-R45))